crates/payload/basic/src/lib.rs

```diff
@@ -204,7 +204,12 @@ where
                 // this only includes changed accounts and storage but is better than nothing
                 let storage =
                     acc.storage.iter().map(|(key, slot)| (*key, slot.present_value)).collect();
-                cached.insert_account(addr, info, storage);
+                cached.insert_account(
+                    addr,
+                    #[allow(clippy::useless_conversion)]
+                    info.into(),
```

Comment:
this conversion causes a loss of information since we convert a `ScrollAccountInfo` to an `AccountInfo`. However, this function is used for payload building in [Ethereum](https://github.com/paradigmxyz/reth/blob/4da6402ea33f931961e3bf54701d27de4fdbe298/crates/ethereum/payload/src/lib.rs#L163-L169). 

When payload building is implemented for Scroll, we might need to udpate the `CachedReads` to provide a `ScrollAccountInfo`. Because Milestone 1 doesn't require payload building, we leave this out for now.